assets/src/stories-editor/components/post-selector/index.js

```diff
@@ -0,0 +1,339 @@
+			selectedSuggestion: null,
+		};
+	}
+
+	componentDidUpdate() {
+		const { showSuggestions, selectedSuggestion } = this.state;
+		if ( showSuggestions && selectedSuggestion !== null && ! this.scrollingIntoView ) {
+			this.scrollingIntoView = true;
+			scrollIntoView( this.suggestionNodes[ selectedSuggestion ], this.autocompleteRef.current, {
+				onlyScrollIfNeeded: true,
+			} );
+
+			this.props.setTimeout( () => {
+				this.scrollingIntoView = false;
+			}, 100 );
+		}
+	}
+
+	componentWillUnmount() {
+		delete this.suggestionsRequest;
+		this.isStillMounted = false;
+	}
+
+	componentDidMount() {
+		this.isStillMounted = true;
+	}
+
+	bindSuggestionNode( index ) {
+		return ( ref ) => {
+			this.suggestionNodes[ index ] = ref;
+		};
+	}
+
+	fetchPostSuggestions( search ) {
+		const searchablePostTypes = this.props.searchablePostTypes || [ 'post' ];
+		const suggestionsRequest = this.suggestionsRequest = apiFetch( {
+			path: addQueryArgs( '/wp/v2/search', {
+				search,
+				per_page: 20,
+				subtype: searchablePostTypes.join( ',' ),
+			} ),
+		} ).then(
+			( suggestions ) => {
+				if ( this.isStillMounted && this.suggestionsRequest === suggestionsRequest ) {
+					this.setState( {
+						suggestions: map( suggestions, ( post ) => ( {
+							id: post.id,
+							title: decodeEntities( post.title ) || __( '(no title)', 'amp' ),
+							postType: post.subtype,
+						} ) ),
+						loading: false,
+					} );
+
+					if ( Boolean( suggestions.length ) ) {
+						this.props.debouncedSpeak( sprintf( _n(
+							'%d result found, use up and down arrow keys to navigate.',
+							'%d results found, use up and down arrow keys to navigate.',
+							suggestions.length, 'amp'
+						), suggestions.length ), 'assertive' );
+					} else {
+						this.props.debouncedSpeak( __( 'No results.', 'amp' ), 'assertive' );
+					}
+				}
+			}
+		).catch(
+			() => {
+				if ( this.isStillMounted && this.suggestionsRequest === suggestionsRequest ) {
+					this.setState( {
+						loading: false,
+					} );
+				}
+			}
+		);
+	}
+
+	updateSuggestions( value ) {
+		// Show the suggestions after typing at least 2 characters.
+		if ( value.length < 2 ) {
+			this.setState( {
+				showSuggestions: false,
+				selectedSuggestion: null,
+				loading: false,
+			} );
+
+			return;
+		}
+
+		this.setState( {
+			showSuggestions: true,
+			selectedSuggestion: null,
+			loading: true,
+		} );
+
+		this.fetchPostSuggestions( value );
+	}
+
+	onChange( event ) {
+		const inputValue = event.target.value;
+		this.props.onChange( inputValue );
+		this.updateSuggestions( inputValue );
+	}
+
+	onKeyDown( event ) {
+		const { showSuggestions, selectedSuggestion, suggestions, loading } = this.state;
+		// If the suggestions are not shown or loading, we shouldn't handle the arrow keys.
+		if ( ! showSuggestions || ! suggestions.length || loading ) {
+			switch ( event.keyCode ) {
+				// When UP is pressed, if the caret is at the start of the text, move it to the 0
```

Comment:
For accessiblity, esc key key press should be handled here, exiting the dropdown. I would also look at tabbing...